proto/rill/runtime/v1/queries.proto

```diff
@@ -0,0 +1,554 @@
+  string metrics_view_name = 2;
+  string dimension_name = 3;
+  repeated string measure_names = 4;
+  google.protobuf.Timestamp time_start = 5;
+  google.protobuf.Timestamp time_end = 6;
+  int64 limit = 7;
+  int64 offset = 8;
+  repeated MetricsViewSort sort = 9;
+  MetricsViewFilter filter = 10;
+  int32 priority = 11;
+}
+
+// Response message for QueryService.MetricsViewToplist
+message MetricsViewToplistResponse {
+  repeated MetricsViewColumn meta = 1;
+  repeated google.protobuf.Struct data = 2;
+}
+
+// Request message for QueryService.MetricsViewTimeSeries
+message MetricsViewTimeSeriesRequest {
+  string instance_id = 1;
+  string metrics_view_name = 2;
+  repeated string measure_names = 3;
+  google.protobuf.Timestamp time_start = 4;
+  google.protobuf.Timestamp time_end = 5;
+  TimeGrain time_granularity = 6;
+  MetricsViewFilter filter = 7;
+  int32 priority = 8;
+}
+
+// Response message for QueryService.MetricsViewTimeSeries
+message MetricsViewTimeSeriesResponse {
+  repeated MetricsViewColumn meta = 1;
+  repeated TimeSeriesValue data = 2;
+}
+
+// Request message for QueryService.MetricsViewTotals
+message MetricsViewTotalsRequest {
+  string instance_id = 1;
+  string metrics_view_name = 2;
+  repeated string measure_names = 3;
+  google.protobuf.Timestamp time_start = 4;
+  google.protobuf.Timestamp time_end = 5;
+  MetricsViewFilter filter = 7;
+  int32 priority = 8;
+}
+
+// Response message for QueryService.MetricsViewTotals
+message MetricsViewTotalsResponse {
+  repeated MetricsViewColumn meta = 1;
+  google.protobuf.Struct data = 2;
+}
+
+// Sort clause for metrics view requests
+message MetricsViewSort {
+  string name = 1;
+  bool ascending = 2;
+}
+
+// Filter clause for metrics view requests
+message MetricsViewFilter {
+  message Cond {
+    string name = 1;
+    repeated google.protobuf.Value in = 2;
+    repeated string like = 3;
+  }
+  repeated Cond include = 2;
+  repeated Cond exclude = 3;
+}
+
+// MetricsViewColumn represents a column in a metrics view
+message MetricsViewColumn {
+  string name = 1;
+  string type = 2;
+  bool nullable = 3;
+}
+
+// **********
+// Profiling APIs
+// **********
+
+message ColumnRollupIntervalRequest {
+  string instance_id = 1;
+  string table_name = 2;
+  string column_name = 3;
+  int32 priority = 4;
+}
+
+message ColumnRollupIntervalResponse {
+  google.protobuf.Timestamp start = 1;
+  google.protobuf.Timestamp end = 2;
+  TimeGrain interval = 3;
+}
+
+message ProfileColumnSummary {
+  oneof case {
+    CategoricalSummary categorical_summary = 1;
+    NumericSummary numeric_summary = 2;
+    TimeRangeSummary time_range_summary = 3;
+  }
+}
+
+// Response for QueryService.ColumnTopK and QueryService.ColumnCardinality. Message will have either topK or cardinality set.
+message CategoricalSummary {
+  oneof case {
+    TopK top_k = 1;
+    double cardinality = 2;
+  }
+}
+
+message ColumnCardinalityResponse {
+  CategoricalSummary categorical_summary = 1;
+}
```

Comment:
Can we have all `...Request` and `...Response` types be defined next to each other (like for `ColumnRollupInterval...`). It makes it much easier to quickly find input and output for an API.